pulsar-client-cpp/include/pulsar/Message.h

```diff
@@ -46,6 +46,16 @@ class Message {
 
     Message();
 
+    /**
+     * Only for MultiTopicsConsumer to get a valid topicName
+     */
+    std::string& getTopicName() const;
```

Comment:
return type should be `const std::string&`

Reply:
In a single topic, we should have the information as well, so we should restrict this method only for the `MultiTopicsConsumer` case

Reply:
Thanks @merlimat.  Currently, as you suggested, move set/getTopicName into private. Is it enough?  e.g. single `ConsumerImpl` could call these 2 methods, but It is not useful at all, maybe no developer will use it inside `ComsumerImpl`. 

Or we could use [this way](https://stackoverflow.com/questions/3220009/is-this-key-oriented-access-protection-pattern-a-known-idiom) to make set/getTopicName only available to MultiTopicsConsumer?